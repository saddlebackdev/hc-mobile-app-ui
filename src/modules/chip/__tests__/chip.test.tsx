// npx jest ./src/modules/chip/__tests__/chip.test.ts

import * as React from 'react';
import {fireEvent, render} from '@testing-library/react-native';

import Chip from '../chip';
import ThemeProvider from '../../theming/theme-provider';
import defaultTheme from '../../theming/default-theme';

describe('Chip', () => {
  describe('Rendering', () => {
    it('renders correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <Chip label="Care" onPress={jest.fn()} />
        </ThemeProvider>,
      );

      const label = wrapper.getByTestId('chip-label');

      expect(label).toBeDefined();

      expect(label.children).toEqual(['Care']);

      expect(wrapper).toMatchSnapshot();
    });

    it('should use passed color as background color', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <Chip label="Care" onPress={jest.fn()} color="#4499ff" />
        </ThemeProvider>,
      );

      const label = wrapper.getByTestId('chip-label');

      expect(label).toBeDefined();

      expect(label.children).toEqual(['Care']);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Interactions', () => {
    it('should call the onPress function on click of "x" button', () => {
      const onPressMock = jest.fn();

      const {getByTestId} = render(
        <ThemeProvider theme={defaultTheme}>
          <Chip label="Care" onPress={onPressMock} />
        </ThemeProvider>,
      );

      const button = getByTestId('chip-close-button');

      expect(button).toBeDefined();

      fireEvent.press(button);

      expect(onPressMock).toHaveBeenCalledTimes(1);
    });
  });
});
