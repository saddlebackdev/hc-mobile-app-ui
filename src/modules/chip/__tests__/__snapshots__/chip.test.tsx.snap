// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip Rendering renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#97A4AB",
        "borderRadius": 20,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 7,
        "paddingVertical": 5,
      },
    ]
  }
  testID="chip"
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "400",
          "textAlign": "left",
        },
      ]
    }
    testID="chip-label"
  >
    Care
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "paddingLeft": 7,
      }
    }
    testID="chip-close-button"
  >
    <Image
      source="test-file-stub"
      style={
        Array [
          Object {
            "height": 16,
            "width": 16,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`Chip Rendering should use passed color as background color 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#4499ff",
        "borderRadius": 20,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 7,
        "paddingVertical": 5,
      },
    ]
  }
  testID="chip"
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "400",
          "textAlign": "left",
        },
      ]
    }
    testID="chip-label"
  >
    Care
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "paddingLeft": 7,
      }
    }
    testID="chip-close-button"
  >
    <Image
      source="test-file-stub"
      style={
        Array [
          Object {
            "height": 16,
            "width": 16,
          },
        ]
      }
    />
  </View>
</View>
`;
